google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/application/AppEngineApplicationCreateDialog.java

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.intellij.appengine.application;
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.services.appengine.v1.model.Location;
+import com.google.cloud.tools.intellij.stats.UsageTrackerProvider;
+import com.google.cloud.tools.intellij.ui.BrowserOpeningHyperLinkListener;
+import com.google.cloud.tools.intellij.util.GctBundle;
+import com.google.cloud.tools.intellij.util.GctTracking;
+
+import com.intellij.openapi.application.ApplicationManager;
+import com.intellij.openapi.application.ModalityState;
+import com.intellij.openapi.progress.ProgressManager;
+import com.intellij.openapi.project.ProjectManager;
+import com.intellij.openapi.ui.DialogWrapper;
+import com.intellij.ui.JBColor;
+
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.awt.Component;
+import java.awt.event.ItemEvent;
+import java.io.IOException;
+import java.util.List;
+
+import javax.swing.JComboBox;
+import javax.swing.JComponent;
+import javax.swing.JPanel;
+import javax.swing.JTextPane;
+
+/**
+ * Dialog that allows creation of an App Engine Application
+ */
+public class AppEngineApplicationCreateDialog extends DialogWrapper {
+
+  private final static String LOCATIONS_DOCUMENTATION_URL
+      = "https://cloud.google.com/docs/geography-and-regions";
+  private final static String FLEX_DOCUMENTATION_URL
+      = "https://cloud.google.com/appengine/docs/flexible/";
+  private final static String STANDARD_DOCUMENTATION_URL
+      = "https://cloud.google.com/appengine/docs/about-the-standard-environment";
+  private static final String HTML_OPEN_TAG = "<html><font face='sans' size='-1'>";
+  private static final String HTML_CLOSE_TAG = "</font></html>";
+
+  private JPanel panel;
+  private JTextPane instructionsTextPane;
+  private JComboBox<AppEngineLocationSelectorItem> regionComboBox;
+  private JTextPane statusPane;
+  private JTextPane regionDetailPane;
+
+  private final Credential userCredential;
```

Comment:
Right now there isn't a service that provides access to a credential in that fashion (that I know of). The credential is originally retrieved from the project selector, since there can be multiple authed users - any project for any of those users is a valid deploy target.

The intended usage pattern here is to dispose of instances of this dialog class once it's closed, and create new instances on demand with the correct credential for a given project.  I'll add some comments that will hopefully clear that up.